tests/utilities/nidcpower_measurement/__init__.py

```diff
@@ -0,0 +1,107 @@
+                voltage_measurements, current_measurements = _source_measure_dc_voltage(
+                    connections, current_limit
+                )
+
+                return (
+                    [session_info.session_name for session_info in session_infos],
+                    [session_info.resource_name for session_info in session_infos],
+                    [session_info.channel_list for session_info in session_infos],
+                    [connection.channel_name for connection in connections],
+                    voltage_measurements,
+                    current_measurements,
+                )
+    else:
+        with measurement_service.context.reserve_session(pin_names) as reservation:
+            with reservation.initialize_nidcpower_session() as session_info:
+                connection = reservation.get_nidcpower_connection(list(pin_names)[0])
+                assert session_info.session is not None
+                voltage_measurements, current_measurements = _source_measure_dc_voltage(
+                    [connection], current_limit
+                )
+
+                return (
+                    [session_info.resource_name],
+                    [session_info.resource_name],
+                    [session_info.channel_list],
+                    [connection.channel_name],
+                    [list(voltage_measurements)[0]],
+                    [list(current_measurements)[0]],
+                )
+
+
+def _source_measure_dc_voltage(
+    connections: Sequence[nims.session_management.TypedConnection[nidcpower.Session]],
+    current_limit: float,
+) -> Tuple[Iterable[float], Iterable[float]]:
+    for connection in connections:
+        channel = connection.session.channels[connection.channel_name]
+        channel.source_mode = nidcpower.SourceMode.SINGLE_POINT
+        channel.output_function = nidcpower.OutputFunction.DC_VOLTAGE
+        channel.current_limit = current_limit
+        channel.voltage_level_range = 10.0
+        channel.current_limit_range = 0.01
+        channel.source_delay = hightime.timedelta(seconds=0.1)
+        channel.voltage_level = 5.0
+
+    voltage_measurements, current_measurements = [], []
+    with ExitStack() as stack:
+        for connection in connections:
+            channel = connection.session.channels[connection.channel_name]
+            stack.enter_context(channel.initiate())
+
+        for connection in connections:
+            channel = connection.session.channels[connection.channel_name]
+            channel.wait_for_event(nidcpower.enums.Event.SOURCE_COMPLETE, timeout=10.0)
+
+        for connection in connections:
+            channel = connection.session.channels[connection.channel_name]
+            measurement = channel.measure_multiple()[0]
+            in_compliance = channel.query_in_compliance()
+            measurement = measurement._replace(in_compliance=in_compliance)
```

Comment:
If you're not going to return in_compliance, then you can delete the call to query it.